### ISSUE TITLE
Wayland: detect the right display even when follow=keyboard/mouse

### ISSUE THREAD

**Issue by fwsmit** [dunst-project/dunst#989]
Now dunst just makes a guess on which display is used or has some workaround. This might cause issues, like maybe #987.

**Comment by BBaoVanC** [comment]
It seems to choose the monitor that has mouse/keyboard focus during startup, and then never change (even if follow is enabled). Is this what you're getting?

**Comment by evgfilim1** [comment]
> It seems to choose the monitor that has mouse/keyboard focus during startup, and then never change (even if follow is enabled). Is this what you're getting?

Looks like I'm getting this. When an app sends a notification, it is displayed on the currently active monitor, but `follow=` is totally ignored.

**Comment by BBaoVanC** [comment]
> > It seems to choose the monitor that has mouse/keyboard focus during startup, and then never change (even if follow is enabled). Is this what you're getting?
> 
> Looks like I'm getting this. When an app sends a notification, it is displayed on the currently active monitor, but `follow=` is totally ignored.

Yeah I'm getting this exact behavior now. It used to only be the monitor that I was on during dunst startup, but now it's whatever monitor I was on when the notification was sent.

**Comment by fwsmit** [comment]
> but now it's whatever monitor I was on when the notification was sent.

That's the intended behavior.

If the notification doesn't follow your cursor, then you should check with your compositor. Dunst sends a hint, but the compositor doesn't have to do anything with it

**Comment by evgfilim1** [comment]
> If the notification doesn't follow your cursor, then you should check with your compositor. Dunst sends a hint, but the compositor doesn't have to do anything with it

But I'm sure it definitely worked earlier

**Comment by fwsmit** [comment]
You can try older versions of sway to see if it works there. I don't think dunst has changed anything here. This isssue is also about something different

**Comment by bynect** [comment]
I think this is the same problem found in #1316
